### PR TITLE
PUBD-652: update MathJax delimiter configuration througout, revise MathJax detect where needed, and upgrade MathJax to v3.2.2

### PR DESCRIPTION
**Pull request by hardyoyo**
- update MathJax delimiter configuration wherever it was specified previously, to omit single dollar signs as an inline math delimiter
- revise MathJax detection heuristic where it is still needed, to use all possible MathJax delimiters
- remove un-needed MathJax detection heuristics
- clean up un-needed MathJax component usage (we now rely entirely on MathJaxContext)
- upgrade to MathJax v. 3.2.2 (which accounts for the large number of changes in this PR)
- has a few style changes to resolve issues with MathJax causing odd bits of truncation

### CODE REVIEW

File: app/jsx/components/TabMainComp.jsx [hunk 1]
```diff
@@ -14,7 +14,6 @@ const config = {
   },
   tex2jax: {
     inlineMath: [
-      ["$", "$"],
```
Comment (myucekul) [inline app/jsx/components/TabMainComp.jsx:17]: Do we still need this config in TabMainComp?

File: app/jsx/components/UnitSeriesComp.jsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import faker from 'faker/locale/en'
 
 class UnitSeriesComp extends React.Component {
   componentDidMount() {
-    $('.c-pub__heading, .c-pub__abstract').dotdotdot({watch: 'window'
+    $('.c-pub__heading, .c-pub__abstract').dotdotdot({watch: 'window', tolerance: 3,
```
Comment (myucekul) [inline app/jsx/components/UnitSeriesComp.jsx:11]: What is the tolerance parameter?

File: app/jsx/pages/PageBase.jsx [hunk 3]
```diff
@@ -26,24 +26,22 @@ const SESSION_EDITING_KEY = "escholEditingPage"
...
     ],
     displayMath: [
       ["$$", "$$"],
       ["\\[", "\\]"]
     ]
   },
-  messageStyle: "none"
+  messageStyle: "none",
+  displayMessages: false,
+  displayErrors: false,
```
Comment (myucekul) [inline app/jsx/pages/PageBase.jsx:41]: I wonder if this config hides problems.

File: app/jsx/pages/PageBase.jsx [hunk 4]
```diff
@@ -26,24 +26,22 @@ const SESSION_EDITING_KEY = "escholEditingPage"
...
+  startup: {
+    typeset: true
```
Comment (myucekul) [inline app/jsx/pages/PageBase.jsx:43]: Isn't typeset=true the default value?

File: app/scss/_pub.scss [hunk 5]
```diff
@@ -27,7 +27,7 @@
 }
 
 .c-pub__abstract {
-  max-height: 8em; // truncate beyond 6 lines per jquery.dotdotdot
+  max-height: 27em; // truncate beyond 15 lines per jquery.dotdotdot
```
Comment (myucekul) [inline app/scss/_pub.scss:30]: Is this as per UX team?

File: app/jsx/components/UnitSeriesComp.jsx (hunk 2 above)
Comment (hardyoyo) [reply app/jsx/components/UnitSeriesComp.jsx:11]: It's described here:
https://dotdotdot.frebsite.nl/#options

I found out about it through a reference on this page:
https://stackoverflow.com/questions/36258246/intermittent-truncation-using-jquery-dotdotdot

It essentially lets the dotdotdot plugin "play nice" with MathJax.

File: app/jsx/components/TabMainComp.jsx (hunk 1 above)
Comment (hardyoyo) [reply app/jsx/components/TabMainComp.jsx:17]: Good catch, we do not.

File: app/jsx/pages/PageBase.jsx [hunk 6]
```diff
@@ -310,7 +308,7 @@ class PageBase extends React.Component
                     fetchingData={this.state.fetchingData}>
           {/* Not sure why the padding below is needed, but it is */}
           <div className="body" style={{ padding: "20px" }}>
-          <MathJaxContext version={3} config={mathjaxConfig} src="/js/MathJax-3.2.2/es5/tex-chtml.js" hideUnilTypeset="every">
+          <MathJaxContext version={3} config={mathjaxConfig} src="/js/MathJax-3.2.0/es5/tex-chtml.js" hideUnilTypeset="every">
```
Comment (myucekul) [inline app/jsx/pages/PageBase.jsx:311]: I'm confused. We are going with MathJax 3.2.2 or not?

**Review by myucekul — APPROVED** [review]

**Pull request merged**